audio/filter/af_scaletempo2_internals.c

```diff
@@ -494,10 +484,8 @@ static void add_input_buffer_final_silence(struct mp_scaletempo2 *p, double play
         return; // no silence needed for iteration
 
     int required_size = needed + p->input_buffer_frames;
-    if (required_size > p->input_buffer_size)
-        resize_input_buffer(p, required_size);
-
     for (int i = 0; i < p->channels; ++i) {
+        MP_TARRAY_GROW(p, p->input_buffer[i], required_size);
```

Comment:
```suggestion
        MP_TARRAY_GROW(p, p->input_buffer[i], required_size - 1);
```

> Resize array p so that p[nextidx] is accessible.

So `size-1` should be enough.

Reply:
That's an interesting quirk of `MP_TARRAY_GROW`; I suppose it's intended to put the same index you're about to write to. I subtracted one from the calculation and renamed the variables to `last_index` to reflect the intention.